pkgs/development/interpreters/dart/default.nix

```diff
@@ -1,39 +1,46 @@
-{ stdenv, fetchurl }:
+{ stdenv, fetchurl, unzip }:
  
-assert stdenv.system == "x86_64-linux" || stdenv.system == "i686-linux";
-
 stdenv.mkDerivation {
-  name = "dart-0.4";
- 
-  installPhase = ''
-    mkdir -p $out
-    cp -R * $out/
-    echo $libPath
-    patchelf --interpreter "$(cat $NIX_CC/nix-support/dynamic-linker)" \
-             --set-rpath $libPath \
-             $out/bin/dart
-    
-    # Hack around weird dart2js resolving bug
-    mv $out/bin/dart2js $out/bin/.dart2js
-    echo "#!/bin/sh" > $out/bin/dart2js
-    echo "$out/bin/.dart2js \$*" >> $out/bin/dart2js
-    chmod +x $out/bin/dart2js
-  '';
-  
+  name = "dart-1.16.1";
```

Comment:
To make updating more convenient, you could add a separate `version` attribute and re-use it both in the `name` and in the source urls below.